src/swap/components/Swap.test.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import { Swap } from './Swap';
+
+jest.mock('wagmi', () => {
+  return {
+    useBalance: jest.fn(),
+    useReadContract: jest.fn(),
+  };
+});
+
+jest.mock('../../internal/text', () => {
+  return {
+    TextTitle3: ({ children }: { children: string }) => <div>{children}</div>,
+  };
+});
```

Comment:
I will fix this in a follow up PR. Getting a bit messy with resolving external dependencies like `clsx`